docs/SERVER-API.md

```diff
@@ -0,0 +1,214 @@
+This document details the API offered by the server to clients. Each method description contains:
+- The method's name
+- The method's purpose and intended effect on the server and/or the appearance of the screen rendered by zellij.
+- The method's parameters
+- The response that the client can expect to receive, including any possible error messages. N.B. Response does not detail side-effects of the request, in particular changes in what's rendered on screen.
+
+The document does not describe implementation details on the server side and is intended for use by those implementing new zellij clients.
+
+
+## Basic controls
+
+### Write input
+
+**Purpose:** Writes a set of bytes to the terminal, not to be interpeted by zellij.
+
+**Parameters:** Vector of bytes to be written.
+
+**Response:** None.
+
+### Switch input mode
+@@@Is this part of the API or a client detail?
```

Comment:
Is it? I think the decision to make is if we want modes to be implemented by all the different clients, or by the server. Could we have both?

If we want it to be implemented just by clients, should we offer a helper library and in what language?

Reply:
I don't know if we really need to have this on the server to be honest. But if we really do I think it would be easier to have them on both sides. Or am I missing something?

Reply:
The plugins run on the server-side. So server needs to be aware of the current input mode.